### ISSUE TITLE
HN 类型 转换出错

### ISSUE THREAD

**Issue by xiaoqiao99** [caj2pdf/caj2pdf#65]
使用的最新版本

错误信息:

```python
    // 软件输出的错误信息
```

请预览一下你填的内容再提交。

**Comment by HinTak** [comment]
@xiaoqiao99 please attach the sample file.

**Comment by HinTak** [comment]
@xiaoqiao99 sorry, cannot download from baidu without baidu account. Please attach using the function below the comment box on github.

**Comment by HinTak** [comment]
The toc seems to be empty (according to linux conversion via libreaderex) but contains garbage according to the python code. Needs code to detect empty toc.

File kept at https://github.com/caj2pdf/CAJSamples/commit/70875e97d101d2f10f0ae0473cd69b7dd2139269

**Comment by Adam-Gao** [comment]
Sorry for the late reply.
If one tries `caj2pdf show`...
```
Type: HN
Page count: 6
Outlines count: 20040
```
20040=0x4E48='HN'

Well, that is obvious now. `caj2pdf` reads file header as `toc_num`.
https://github.com/caj2pdf/caj2pdf/blob/ea7a260a778c4793a17285ec1a9dcebd4284e0a9/cajparser.py#L31-L38

`self._TOC_NUMBER_OFFSET = 0` means there is NO TOC, rather than to use file header (OFFSET 0) as `toc_num`.

The code is quite confusing and logically buggy. Maybe I will try to fix this when I am free.

**Comment by HinTak** [comment]
Argh, elsewhere I had some "if (self._TOC_NUMBER_OFFSET > 0):" . the outline code needs to generate an empty outline, which is slightly more work than just return.

**Comment by HinTak** [comment]
I have added a few "if (self._TOC_NUMBER_OFFSET > 0):" . Now it finishes, but the size of the pages/images are wrong so the pdf does not display correctly. Somehow the pages are all 3084 x 3084 pts in size (a very large number). If somebody can find and fix the page size, it should work.

There are two images in the pdf, the 2nd one is a full page one. The first page still won't be entirely correct (missing text), but that's another issue.

**Comment by Adam-Gao** [comment]
Well, I underestimated this issue. Six pages and two images, the sample seems like a partial pure-text HN which we cannot handle with at this time.

@JeziL is right.
https://github.com/caj2pdf/caj-convert/issues/1#issuecomment-883412715

It seems like the file is produced by some typography specialized software commonly used in Beijing... I have no idea how to cope with these files.

**Comment by HinTak** [comment]
Yes, it is one of those partially pure-text HN files. The "pure-text" description is not entirely correct - that data is positioning + text. I have started striping the positioning part in HNParsePage.py, but potentially one can recreate the page, if one gets at the positioning correctly plus correct text too. This is do-able, now that we have native/genuine conversion to compare with. It is just not a quick thing to achieve.

**Comment by mpr1255** [comment]
FWIW I have been having some success with using the `parse` function, then cleaning up the output (sed etc.), and then passing it to chatgpt (3.5-turbo - lots cheaper) to clean up. It _appears_ that fully formed text is there -- it's just that it's buried. Just cleaning the raw text out then formatting it nicely seems to work.
